/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Use Sun branding colors in various hacks to better match color schemes of JDS
desktop.

diff -urp -x '*~' driver/XScreenSaver.ad.in driver/XScreenSaver.ad.in
--- driver/XScreenSaver.ad.in	2006-09-19 17:09:08.203402000 -0700
+++ driver/XScreenSaver.ad.in	2006-09-19 17:09:13.618387000 -0700
@@ -293,7 +293,7 @@ GetViewPortIsFullOfLies: False
                                 bubbles -root				    \n\
 				pong -root				    \n\
 				wormhole -root				    \n\
-				fuzzyflakes -root			    \n\
+				fuzzyflakes -root -color A3B8CB		    \n\
 				anemotaxis -root			    \n\
 				memscroller -root			    \n\
 				substrate -root				    \n\
diff -urp -x '*~' hacks/glx/antinspect.c hacks/glx/antinspect.c
--- hacks/glx/antinspect.c	2006-02-25 12:13:26.000000000 -0800
+++ hacks/glx/antinspect.c	2006-09-19 17:09:13.619018000 -0700
@@ -94,13 +94,13 @@ static const float position1[] = {-1.0, 
 static const float lmodel_ambient[] = {0.5, 0.5, 0.5, 1.0};
 static const float lmodel_twoside[] = {GL_TRUE};
 
-static const float MaterialRed[] =     {0.6, 0.0, 0.0, 1.0};
-static const float MaterialOrange[] =  {1.0, 0.69, 0.00, 1.0};
-static const float MaterialGray[] =    {0.2, 0.2, 0.2, 1.0};
+static const float MaterialRed[] =     {0.90, 0.43, 0.0, 1.0};
+static const float MaterialOrange[] =  {1.0, 0.78, 0.15, 1.0};
+static const float MaterialGray[] =    {0.2, 0.21, 0.22, 1.0};
 static const float MaterialBlack[] =   {0.1, 0.1, 0.1, 0.4};
 static const float MaterialShadow[] =   {0.3, 0.3, 0.3, 0.3};
-static const float MaterialGray5[] =   {0.5, 0.5, 0.5, 0.3};
-static const float MaterialGray6[] =   {0.6, 0.6, 0.6, 1.0};
+static const float MaterialGray5[] =   {0.5, 0.51, 0.53, 0.3};
+static const float MaterialGray6[] =   {0.6, 0.61, 0.63, 1.0};
 
 static antinspectstruct *antinspect = (antinspectstruct *) NULL;
 
diff -urp -x '*~' hacks/glx/antmaze.c hacks/glx/antmaze.c
--- hacks/glx/antmaze.c	2006-03-14 19:55:05.000000000 -0800
+++ hacks/glx/antmaze.c	2006-09-19 17:09:13.619973000 -0700
@@ -148,13 +148,13 @@ typedef struct {
 static antmazestruct *antmaze = (antmazestruct *) NULL;
 
 
-static const GLfloat MaterialRed[] = {0.6, 0.0, 0.0, 1.0};
+static const GLfloat MaterialRed[] = {0.75, 0.40, 0.0, 1.0};
 /*static const GLfloat MaterialMagenta[] = {0.6, 0.2, 0.5, 1.0};*/
 static const GLfloat MaterialGray8[] = {0.8, 0.8, 0.8, 1.0};
-static const GLfloat MaterialGray35[] = {0.30, 0.30, 0.30, 1.0};
-static const GLfloat MaterialGray4[] = {0.40, 0.40, 0.40, 1.0};
-static const GLfloat MaterialOrange[] = {1.0, 0.69, 0.00, 1.0};
-static const GLfloat MaterialGreen[] = {0.1, 0.4, 0.2, 1.0};
+static const GLfloat MaterialGray35[] = {0.30, 0.31, 0.33, 1.0};
+static const GLfloat MaterialGray4[] = {0.4375, 0.4450, 0.4650, 1.0};
+static const GLfloat MaterialOrange[] = {0.90, 0.43, 0.00, 1.0};
+static const GLfloat MaterialGreen[] = {0.695, 0.734, 0.0, 1.0};
 
 /* lighting variables */
 static const GLfloat front_shininess[] = {60.0};
diff -urp -x '*~' hacks/glx/gears.c hacks/glx/gears.c
--- hacks/glx/gears.c	2006-03-12 03:15:36.000000000 -0800
+++ hacks/glx/gears.c	2006-09-19 17:09:13.624041000 -0700
@@ -640,10 +640,10 @@ pinit(ModeInfo * mi)
 {
 	gearsstruct *gp = &gears[MI_SCREEN(mi)];
 	static const GLfloat pos[4]   = {5.0, 5.0, 10.0, 1.0};
-	static const GLfloat red[4]   = {0.8, 0.1, 0.0, 1.0};
-	static const GLfloat green[4] = {0.0, 0.8, 0.2, 1.0};
-	static const GLfloat blue[4]  = {0.2, 0.2, 1.0, 1.0};
-	static const GLfloat gray[4]  = {0.5, 0.5, 0.5, 1.0};
+	static const GLfloat red[4]   = {0.9023, 0.4336, 0.0, 1.0};
+	static const GLfloat green[4] = {0.6953, 0.7344, 0.0, 1.0};
+	static const GLfloat blue[4]  = {0.3242, 0.5078, 0.6289, 1.0};
+	static const GLfloat gray[4]  = {0.4375, 0.4450, 0.4650, 1.0};
 	static const GLfloat white[4] = {1.0, 1.0, 1.0, 1.0};
 	int         wire = MI_IS_WIREFRAME(mi);
 	int         mono = MI_IS_MONO(mi);
diff -urp -x '*~' hacks/glx/sproingies.c hacks/glx/sproingies.c
--- hacks/glx/sproingies.c	2006-09-18 14:45:17.000000000 -0700
+++ hacks/glx/sproingies.c	2006-09-19 17:09:13.657559000 -0700
@@ -113,9 +113,9 @@ build_TopsSides(int wireframe)
 
 	/* Surface: Tops */
 	glNewList(dl_num, GL_COMPILE);
-	mat_color[0] = 0.392157;
-	mat_color[1] = 0.784314;
-	mat_color[2] = 0.941176;
+	mat_color[0] = 0.325490;
+	mat_color[1] = 0.509804;
+	mat_color[2] = 0.631373;
 	if (wireframe)
 		glColor3fv(mat_color);
 	else {
@@ -129,9 +129,9 @@ build_TopsSides(int wireframe)
 		glColor3fv(mat_color);
 	else {
       /* jwz: in wireframe mode, color tops and sides the same. */
-      mat_color[0] = 0.156863;
-      mat_color[1] = 0.156863;
-      mat_color[2] = 0.392157;
+		mat_color[0] = 0.207843;
+		mat_color[1] = 0.333333;
+		mat_color[2] = 0.419607;
 		glMaterialfv(GL_FRONT, GL_AMBIENT_AND_DIFFUSE, mat_color);
 	}
 	glEndList();
diff -urp -x '*~' utils/colors.c utils/colors.c
--- utils/colors.c	2006-02-04 05:39:41.000000000 -0800
+++ utils/colors.c	2006-09-19 17:09:13.658183000 -0700
@@ -598,6 +598,13 @@ make_uniform_colormap (Display *dpy, Vis
   *ncolorsP = ncolors;
 }
 
+/* Use Sun branding colors first in random colormaps */
+#define NUM_SUN_COLORS 12
+static const XColor sun_colors[NUM_SUN_COLORS] = {
+    { 83, 130, 161}, {231, 111,  0}, {178, 188,   0}, {255, 199,  38},
+    { 53,  85, 107}, {192, 102,  0}, {127, 120,   0}, {198, 146,   0},
+    {163, 184, 203}, {237, 155, 79}, {197, 213, 169}, {248, 213, 131}
+};
 
 void
 make_random_colormap (Display *dpy, Visual *visual, Colormap cmap,
@@ -610,6 +617,7 @@ make_random_colormap (Display *dpy, Visu
   Bool wanted_writable = (allocate_p && writable_pP && *writable_pP);
   int ncolors = *ncolorsP;
   int i;
+  int sun_color_offset = random() % NUM_SUN_COLORS;
   Screen *screen = (dpy ? DefaultScreenOfDisplay(dpy) : 0); /* #### WRONG! */
 
   if (*ncolorsP <= 0) return;
@@ -631,9 +639,16 @@ make_random_colormap (Display *dpy, Visu
 	}
       else
 	{
+	  if (i < NUM_SUN_COLORS) {
+	      int n = (i + sun_color_offset) % NUM_SUN_COLORS;
+	      colors[i].red = sun_colors[n].red;
+	      colors[i].green = sun_colors[n].green;
+	      colors[i].blue = sun_colors[n].blue;
+	  } else {
 	  colors[i].red   = random() % 0xFFFF;
 	  colors[i].green = random() % 0xFFFF;
 	  colors[i].blue  = random() % 0xFFFF;
+	  }
 	}
     }
 
